api/tests/Behat/bootstrap/v2/Common/FormFillingTrait.php

```diff
@@ -282,21 +294,31 @@ public function editAnswerInSectionTrackTotal(NodeElement $summaryRowToEdit, str
      * @param string      $newValue         The new value
      * @param string      $formSectionName  Which section name in $submittedAnswersByFormSections the item to
      *                                      edit belongs to
+     * @param string      $type             Whether it's a option or a field that needs changing
+     * @param string|null $translatedValue  The translated value if option. If field leave blank
      *
-     * @throws ElementNotFoundException
+     * @throws BehatException
      */
-    public function editAnswerInSection(NodeElement $summaryRowToEdit, string $fieldName, string $newValue, string $formSectionName): array
+    public function editAnswerInSection(NodeElement $summaryRowToEdit, string $fieldName, $newValue, string $formSectionName, bool $fullGroup = true, string $type = 'field', string $translatedValue = null)
     {
-        $this->removeAnswerFromSection($fieldName, $formSectionName);
-
-        $summaryRowToEdit->clickLink('Edit');
```

Comment:
@acsauk Sorry about this. Just could not get the selection to work properly. Was pulling out my hair so created a helper and changed this to click. Hopefully that's ok or if not maybe you can show me where i was going wrong. Spent over an hour mucking around with that one bit of syntax :( 
This still works with your existing code.

Reply:
I've just had to deal with this in my PR too. While it's nice to have a single function I think having to work out what 7 arguments relate too and use around the codebase might be a step too far. I broke this out to a separate function to be explicit about using it for a select - https://github.com/ministryofjustice/opg-digideps/pull/660/files#diff-62d6fee126d7c272fc2598fcbe612644e274ea3149cea9b44cec999ec9f70285. What do you think?

Reply:
Yeah can't argue with that. I have a penchant for sticking things in one function that ought to be in two. It's a bad habit!